tests/unit/Command/FixEncryptedVersionTest.php

```diff
@@ -0,0 +1,173 @@
+		$view = new View("/" . self::TEST_ENCRYPTION_VERSION_AFFECTED_USER . "/files");
+
+		$view->touch('hello.txt');
+		$view->touch('world.txt');
+		$view->file_put_contents('hello.txt', 'a test string for hello');
+		$view->file_put_contents('world.txt', 'a test string for world');
+
+		$fileInfo = $view->getFileInfo('hello.txt');
+
+		$storage = $fileInfo->getStorage();
+		$cache = new Cache($storage);
+		$fileCache = $cache->get($fileInfo->getId());
+
+		//Now change the encrypted version to zero
+		$cacheInfo = ['encryptedVersion' => 0, 'encrypted' => 0];
+		$cache->put($fileCache->getPath(), $cacheInfo);
+
+		$this->commandTester->execute([
+			'user' => self::TEST_ENCRYPTION_VERSION_AFFECTED_USER
+		]);
+
+		$output = $this->commandTester->getDisplay();
+		$this->assertEquals("Verifying the content of file /test_enc_version_affected_user1/files/hello.txt
+Increment the encrypted version to 1
+The file /test_enc_version_affected_user1/files/hello.txt is: OK
+Fixed file /test_enc_version_affected_user1/files/hello.txt with version 1
+Verifying the content of file /test_enc_version_affected_user1/files/welcome.txt
+The file /test_enc_version_affected_user1/files/welcome.txt is: OK
+Verifying the content of file /test_enc_version_affected_user1/files/world.txt
+The file /test_enc_version_affected_user1/files/world.txt is: OK
+", $output);
+	}
+
+	/**
+	 * In this test the encrypted version of the file is less than the original value
+	 * but greater than zero
+	 */
+	public function testEncryptedVersionLessThanOriginalValue() {
+		$this->markTestSkipped("Skipping this test because the user login would fail in the CI. And once we have a db endpoints available in acceptance test, we would add the test over there");
+		\OC::$server->getUserSession()->login(self::TEST_ENCRYPTION_VERSION_AFFECTED_USER, 'foo');
+		$view = new View("/" . self::TEST_ENCRYPTION_VERSION_AFFECTED_USER . "/files");
+
+		$view->touch('hello.txt');
+		$view->touch('world.txt');
+		$view->file_put_contents('hello.txt', 'a test string for hello');
+		$view->file_put_contents('hello.txt', 'Yet another value');
+		$view->file_put_contents('hello.txt', 'Lets modify again');
+		$view->file_put_contents('world.txt', 'a test string for world');
+
+		$fileInfo = $view->getFileInfo('hello.txt');
+
+		$storage = $fileInfo->getStorage();
+		$cache = new Cache($storage);
+		$fileCache = $cache->get($fileInfo->getId());
+
+		//Now change the encrypted version to zero
+		$cacheInfo = ['encryptedVersion' => 2, 'encrypted' => 2];
+		$cache->put($fileCache->getPath(), $cacheInfo);
+
+		$this->commandTester->execute([
+			'user' => self::TEST_ENCRYPTION_VERSION_AFFECTED_USER
+		]);
+
+		$output = $this->commandTester->getDisplay();
+		$this->assertEquals("Verifying the content of file /test_enc_version_affected_user1/files/hello.txt
+Decrement the encrypted version to 1
+Increment the encrypted version to 3
+Increment the encrypted version to 4
```

Comment:
please test with further increments as I seem to have spotted a bug in the increment logic... see my other comment above

Reply:
Added further increments in the same test.